src/librustc_codegen_llvm/declare.rs

```diff
@@ -162,6 +162,15 @@ pub fn define_global(cx: &CodegenCx, name: &str, ty: Type) -> Option<ValueRef> {
     }
 }
 
+/// Declare a private global
+///
+/// Use this function when you intend to define a global without a name.
+pub fn define_private_global(cx: &CodegenCx, ty: Type) -> ValueRef {
+    unsafe {
+        llvm::LLVMRustGetOrInsertGlobal(cx.llmod, ptr::null(), ty.to_ref())
```

Comment:
This should probably also set the appropriate linkage, so that the users of this function don’t need to do that.